### ISSUE TITLE
[TODO] Investigate equivalence tests

### ISSUE THREAD

**Issue by ydshieh** [huggingface/transformers#16497]
Currently the PT/TF/Flax equivalence tests use `1e-5` as the tolerance for the absolute differences of outputs.

We see that these tests failed with a non-negligible (although not carefully defined) frequency.

Create this page to track a list of models to investigate.

- **FlaxWav2Vec2ModelTest** (2.2888184e-05 > 1e-5)
  - https://app.circleci.com/pipelines/github/huggingface/transformers/37363/workflows/a4b06424-0ba8-4fbc-9054-6ff52fbf8145/jobs/411654 

- **TFGPT2EncoderDecoderModelTest** (0.001009281724691391 > 1e-3)
  - https://app.circleci.com/pipelines/github/huggingface/transformers/37358/workflows/43c12161-33d8-4df5-ba3c-3e62a4507ee7/jobs/411579

- **TFViTMAEModelTest** (1.013279e-05 > 1e-5)
  - https://app.circleci.com/pipelines/github/huggingface/transformers/37319/workflows/5adfba7a-d12b-4e1e-9a7a-e33c7d5fd6ee/jobs/411002

**Comment by gante** [comment]
Another one to add to this list: `tests/funnel/test_modeling_funnel.py::FunnelModelTest::test_pt_tf_model_equivalence`. I've been getting a failure in this one every other day -- example: https://app.circleci.com/pipelines/github/huggingface/transformers/38007/workflows/2a98b7b1-5ad0-4b80-a702-1887c620193f/jobs/421265

**Comment by ydshieh** [comment]
> Another one to add to this list: `tests/funnel/test_modeling_funnel.py::FunnelModelTest::test_pt_tf_model_equivalence`. I've been getting a failure in this one every other day -- example: https://app.circleci.com/pipelines/github/huggingface/transformers/38007/workflows/2a98b7b1-5ad0-4b80-a702-1887c620193f/jobs/421265

Thanks. @stas00 also reported this. I will take a look~

**Comment by ydshieh** [comment]
> Another one to add to this list: `tests/funnel/test_modeling_funnel.py::FunnelModelTest::test_pt_tf_model_equivalence`. I've been getting a failure in this one every other day -- example: https://app.circleci.com/pipelines/github/huggingface/transformers/38007/workflows/2a98b7b1-5ad0-4b80-a702-1887c620193f/jobs/421265

(just for the record) 34 / 500 runs have `FunnelForMaskedLM.output.logits` at around `1e-5` ~ `2e-5`: so ~ `6.8%` chance of failure 😢

**Comment by gante** [comment]
@ydshieh I believe you can add the `WIP` label to stop the bot :)

**Comment by ydshieh** [comment]
> 

I am afraid I will completely forget this issue 😸. But if this brother you guys, I can do it. Thanks for the tip, I didn't know about it
